### ISSUE TITLE
How to train a detection model when image width is not equal to image height?

### ISSUE THREAD

**Issue by retioa11** [ultralytics/ultralytics#4832]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

The width and height of the detected image are [1920, 1080], not equal. And I want to train a *.pt model and its input image size is [960, 544]. Is it feasible? Is it reasonable?
I set imgsz=[960, 544], epochs=3, device=0, rect=True for Training，But the imgsz parameter in the pt file is still 640.  How can I make it a [960,544]?
![image](https://github.com/ultralytics/ultralytics/assets/81463532/3fdd84a3-f2b9-4356-ad09-205c3fcb7475)
![image](https://github.com/ultralytics/ultralytics/assets/81463532/7fd73b61-7add-49da-bd31-20bd54b22260)

**Comment by glenn-jocher** [comment]
@RizwanMunawar hey I just had an idea, maybe we should default `rect=True` when a user passes an imgsz of 2 values?

**Comment by retioa11** [comment]
> @retioa11! I've used the mentioned command to perform model fine-tuning on the custom dataset with image sizes set to 960x544, and it's functioning as expected.
> 
> ```
> yolo task=detect mode=train model=yolov8n.pt imgsz=960,544 rect=True
> ```
> 
> **More Information**
> 
> ![imgsz_test](https://user-images.githubusercontent.com/62513924/266998574-2162a54e-3bfb-4471-8c13-84ecd204205b.png)
Thank you。But how to make the image size in the pt model file [960,544]? The size obtained using the following code is [960,960].
 
`    entrypoint(debug='yolo task=detect mode=train model=F:/33DiscernAccelerate/Code/AITool/3.ModelTrain/yolov8n.pt'
                     ' imgsz=960,544 rect=True epochs=11')`

![image](https://github.com/ultralytics/ultralytics/assets/81463532/18dfecb6-9297-4238-83ea-08741ea24937)
![image](https://github.com/ultralytics/ultralytics/assets/81463532/e7e55e6e-afd3-4a4e-a81b-3aacb159e16e)

**Comment by glenn-jocher** [comment]
@retioa11 the `imgsz` argument specifies the target resolution that your images will be resized to during training. These dimensions are stored in the model file (.pt). The `imgsz` of [960, 544] indicates a width of 960 and a height of 544. Using `rect=True` enables rectangular training, which can provide potential improvements to accuracy and speed, and should correctly set the dimensions.

However, please note that the output resolution in the .pt file is always a square. This is because the architecture takes into consideration the highest value among both to keep the network symmetrical, in this case, 960. So, even though you trained the model with images of size [960,544], the architecture will have an effective input size of [960,960]. And during detection, the input images will be scaled maintaining the aspect ratios into a canvas of [960,960]. So, don't worry, your model is still effectively trained on your desired resolution, and it should work fine for inference on your target images.

**Comment by RizwanMunawar** [comment]
> @RizwanMunawar hey I just had an idea, maybe we should default `rect=True` when a user passes an imgsz of 2 values?

@glenn-jocher Yes as you mentioned in the last comment, the output resolution of the pt file is always a square, so I think it is better to set rect=True default.

**Comment by retioa11** [comment]
@glenn-jocher Thank you。

**Comment by glenn-jocher** [comment]
@retioa11 you're welcome! If you have any more questions or require further assistance, feel free to reach out. Happy training with YOLOv8!
